src/testRunner/unittests/sys/symlinkWatching.ts

```diff
@@ -0,0 +1,222 @@
+                            deferred[indexForDefer++].resolve();
+                        },
+                        10,
+                        watchOptions,
+                    ),
+                    deferred,
+                };
+            }
+        });
+    }
+
+    function verifyWatchDirectoryUsingFsEvents<System extends ts.System>(
+        sys: System,
+        dir: string,
+        link: string,
+        fsWatch: (dir: string, cb: ts.FsWatchCallback, sys: System) => ts.FileWatcher,
+    ) {
+        it(`watchDirectory using fsEvents`, async () => {
+            const expectedEvent = ["rename", "change", "rename", "change"];
+            const expectedFileName = ["file1.ts", "file1.ts", "file2.ts", "file2.ts"];
+            const fileResult = watchDirectory(dir);
+            const linkResult = watchDirectory(link);
+            delayedOp(() => sys.writeFile(`${dir}/file1.ts`, "export const x = 100;"));
+
+            // Should invoke on file as well as link, rename and change
+            await fileResult.deferred[0].promise;
+            await linkResult.deferred[0].promise;
+            await fileResult.deferred[1].promise;
+            await linkResult.deferred[1].promise;
+
+            delayedOp(() => sys.writeFile(`${link}/file2.ts`, "export const x = 100;"));
+            // // Should invoke on file as well as link, rename and change
+            await fileResult.deferred[2].promise;
+            await linkResult.deferred[2].promise;
+            await fileResult.deferred[3].promise;
+            await linkResult.deferred[3].promise;
+
+            fileResult.watcher.close();
+            linkResult.watcher.close();
+
+            function watchDirectory(dir: string) {
+                const deferred = [defer(), defer(), defer(), defer()];
+                let indexForDefer = 0;
+                return {
+                    dir,
+                    watcher: fsWatch(
+                        dir,
+                        (event, fileName) => {
+                            assert.equal(event, expectedEvent[indexForDefer]);
+                            assert(!fileName || fileName === expectedFileName[indexForDefer]);
+                            deferred[indexForDefer++].resolve();
+                        },
+                        sys,
+                    ),
+                    deferred,
+                };
+            }
+        });
+    }
+
+    function getFileName(): (dir: string) => string {
+        return dir => `${dir}/${ts.getBaseFileName(dir)}.ts`;
+    }
+
+    describe("with ts.sys::", () => {
+        const root = ts.normalizePath(IO.joinPath(IO.getWorkspaceRoot(), "tests/baselines/symlinks"));
+        before(() => {
+            cleanup();
+            ts.sys.writeFile(`${root}/polling/file.ts`, "export const x = 10;");
+            withSwallowException(() => fs.symlinkSync(`${root}/polling`, `${root}/linkedpolling`, "junction"));

```

Comment:
Does this test need to be using `symlink` on non-windows OSs?

Reply:
i am not sure what you mean. This code becomes OS independent with the usage of symlinkSync

Reply:
I guess it only works because all OSs besides Windows ignore the symlink type parameter. Feels sorta strange.

I will say that I'm wondering if we should get #52350 in so that we're actually running the tests on Windows; right now we're just testing Linux so can't verify that this PR is okay elsewhere.

Reply:
can we do this sooner then.. i would like to get this change in so that we are consistent in the watch behaviour testing. 
I have some follow up issues that need this to be merged

Reply:
Merged that PR, so you should be good to merge main or open/close to rerun with the new jobs.